tests/test_recipients_viewsets.py

```diff
@@ -0,0 +1,129 @@
+from unittest.mock import Mock
+
+import pytest
+
+from django.core.cache import caches
+from django.urls import reverse
+
+from wagtail_newsletter.campaign_backends.mailchimp import MailchimpCampaignBackend
+from wagtail_newsletter.viewsets import Audience, AudienceSegment
+
+
+class MockMailchimpCampaignBackend(MailchimpCampaignBackend):
+    def __init__(self):
+        self.client = Mock()
+
+
+@pytest.fixture(autouse=True)
+def memory_cache(settings):
+    settings.CACHES = {
+        "default": {"BACKEND": "django.core.cache.backends.locmem.LocMemCache"},
+    }
+    caches["default"].clear()
+
+
+@pytest.fixture(autouse=True)
+def backend(monkeypatch: pytest.MonkeyPatch):
+    backend = MockMailchimpCampaignBackend()
+
+    backend.client.lists.get_all_lists.return_value = {
+        "lists": [
+            {"id": "be13e6ca91", "name": "Torchbox", "stats": {"member_count": 8}},
+            {"id": "9af08f2afa", "name": "Other", "stats": {"member_count": 13}},
+        ],
+    }
+
+    backend.client.lists.list_segments.return_value = {
+        "list_id": "be13e6ca91",
+        "segments": [
+            {"id": 2103836, "member_count": 3, "name": "Segment One"},
+            {"id": 2103837, "member_count": 1, "name": "Segment Two"},
+            {"id": 2103838, "member_count": 1, "name": "Segment Three"},
+        ],
+    }
+
+    monkeypatch.setattr(
+        "wagtail_newsletter.campaign_backends.get_backend",
+        lambda: backend,
+    )
+    return backend
```

Comment:
maybe something like this would be better, as it would ensure that the mock is cleared after the test?

```py
with patch("wagtail_newsletter.campaign_backends.get_backend") as get_backend:
    get_backend.return_value = backend
    yield backend
# after the test we automatically exit the `with` context, 
# which restores the original function :-)

Reply:
TIL! :smile: